content/en/blog/2022-08-05-effective-programmers-read-code.md

```diff
@@ -0,0 +1,59 @@
+
+The first and most important notion is that these audiences are themselves dynamic. The more time that passes from the original implementation of the source code, the more different both the runtime and the reader are. We should write code to ensure longevity of executability and understandability.
+
+When it comes to writing code, we certainly should not only focus on or prioritize execution performance if it sacrifices readability. Self documenting code is absolutely something to strive for. On the other hand, clean coders eschew comments and have probably have taken this to the extreme. Obtuse code serves no one, not future readers and not compilers. Verbose comments that repeat the code make it harder to build the glass castle. Instead, use comments to create markers or way points that help guide the reader to other pieces of important or related code, or that serve to build the glass castle more efficiently.
+
+Documentation is essential and should give us a high level overview of what the code should do. The best documentation gives us entry points into the code so that we can understand it further, but doesn't go so far that it prevents you from engaging with the code itself. Ensure that all packages, structs, methods, and functions are correctly documented with doc strings and that all are reachable from the map that the documentation describes.
+
+
+[^1]: My inspiration for the term "glass castle" comes from the ["mind palace"](https://www.smithsonianmag.com/arts-culture/secrets-sherlocks-mind-palace-180949567/) portrayed by Benedict Cumberbach as Sherlock Holmes in BBC's _Sherlock_ (2010). Sherlock uses the mind palace as a memory technique where memories are linked to objects which are placed in specific places in the palace. I thought "palace" was a little too grandiose for what I did so I changed it to castle and modified it with "glass" to imply the fragility of the mental abstraction while programming. I suppose "mind glass castle" would be the full term but that was a little unwieldy.
+
+[^2]: My experience of the glass castle is as a graph or molecular structure which I can "see" in my mind. In fact, sight is the best way to describe my interaction with the castle sort of like how reading instead of hearing is the best way to describe how I consume an audiobook. It's a bit hard to describe what I see, just as it would be hard to describe a dream, but I think the important note here is that as you program, you should be mentally interacting with this model in a way that makes sense to you, e.g. is it "touch" or "movement" for you?
+
+[^3]: And that is before code reviews!
```

Comment:
This was a really cool perspective and read! The suggestions I had were all grammar/clarity-related...the content looks great to me:)

Reply:
Per the image, are we still keeping with the 'nature' theme of previous blog images? If so, the one I'm attaching below might be an option? (an example of 'reading' nature
![BN-WO094_bkrvna_M_20171212115450](https://user-images.githubusercontent.com/100060648/183275267-cfa97454-2352-44ce-a7f6-33b97001189c.jpeg)
)

Reply:
We loved this image but realized we couldn't use it with the copyright permissions 😞, so I've updated the image to use another one with a similar (I hope) vibe 🌳